### PR TITLE
Icons 2

### PR DESCRIPTION
**Pull request by mridulnagpal**
@jvalleroy I made a new PR as the other one was getting messy. In this all the icons are added in the static folder and seem to run when I deleted that static folder from vagrant ssh. Also changed the xmpp icon as @mray suggested. Please have a look.
#670 #594

### CODE REVIEW

**Comment by jvalleroy** [comment]
@mridulnagpal I like the appearance overall. Note that the ikiwiki icon doesn't work though (due to the shortcut ID for wikis).

Also we will need to list each image file in https://github.com/freedombox/Plinth/blob/master/LICENSES.

**Comment by mridulnagpal** [comment]
@jvalleroy Please have a look.

**Comment by mray** [comment]
* Some logos look too big to me. We are only going to use them on the web interface at relatively low resolutions. Even though we have them locally it would make sense to keep file size low, as it is more resource saving (performance). Not sure where the sweet spot should be, my guess is something below 500px

* jsxc.png should probably only be cropped to the acronym letters
image

* ejabberd should be just the "e"
uwam5hhj_400x400

**Comment by mridulnagpal** [comment]
@jvalleroy Every icon is now below 500px and both these icons have been
replaced

**Comment by jvalleroy** [comment]
@mridulnagpal In LICENSES, we need to list the actual license for each file and a reference to verify the license. For example:
```
- static/themes/default/icons/deluge.png :: [[https://en.wikipedia.org/wiki/Deluge_(software)#/media/File:Deluge-Logo.svg][GPL]]
```

**Comment by mridulnagpal** [comment]
@jvalleroy I know the links to the images how do I get the license??

**Comment by jvalleroy** [comment]
@mridulnagpal You would just have to search for each one. Maybe try to look up each project, they should have a repository containing the icon and also a license.

For example I found the minetest icon here: https://github.com/minetest/minetest/tree/master/misc
Then I found the license for the repository at https://github.com/minetest/minetest/blob/master/README.txt#L426 which says it's LGPL 2.1 or later.

**Comment by mray** [comment]
jsxc probably has no licence - it is just a screnshot of a canned github page theme.

**Comment by mridulnagpal** [comment]
@jvalleroy @mray For mumble the License says:

Copyright © The author

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

   1. Redistributions of source code must retain the above copyright
   notice, this list of conditions and the following disclaimer.
   2. Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
   3. Neither the name of The author nor the names of its contributors may
   be used to endorse or promote products derived from this software without
   specific prior written permission.

This software is provided by The author and contributors "as is" and any
express or implied warranties, including, but not limited to, the implied
warranties of merchantability and fitness for a particular purpose are
disclaimed. In no event shall The author and contributors be liable for any
direct, indirect, incidental, special, exemplary, or consequential damages
(including, but not limited to, procurement of substitute goods or
services; loss of use, data, or profits; or business interruption) however
caused and on any theory of liability, whether in contract, strict
liability, or tort (including negligence or otherwise) arising in any way
out of the use of this software, even if advised of the possibility of such
damage.


What do I write??

**Comment by jvalleroy** [comment]
@mridulnagpal That's the BSD-3-clause. There's a list of common licenses at https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/.

**Comment by mridulnagpal** [comment]
@jvalleroy Updated except for jsxc

**Comment by jvalleroy** [comment]
@mridulnagpal This seems to be the actual logo for Privoxy: https://openhub.net/p/privoxy
Although I'm not sure about the source, but it looks similar to the favicon for privoxy.org.

The other logo is on http://vpnpick.com/using-hamachi-privoxy-home-vpn-solution/, I see a copyright notice but no license, so I'm not sure if we can actually use that one. I think it would safer to use the one on openhub.net.

Sorry if this is a bit tedious :)

**Comment by mridulnagpal** [comment]
@jvalleroy Done :)

**Comment by mridulnagpal** [comment]
@jvalleroy I looked at the previous version on its website. Corrected to
GPL-3+
Clarification and Preparation for the Future

Because of the incorporated numerous other components, the only options for
licenses are in the GNU GPL family, namely GNU GPLv2, GNU GPLv3, GNU AGPLv3.

File: plinth/modules/deluge/__init__.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 managed_packages = ['deluged', 'deluge-web']
 
-title = _('BitTorrent Web Client (Deluge)')
+title = ('BitTorrent Web Client \n (Deluge)')
```
Comment (jvalleroy) [inline plinth/modules/deluge/__init__.py]: The _ in front of the string is important. It's an alias for ugettext_lazy, which we need for translations to work.

**Comment by mridulnagpal** [comment]
the \n wasn't working in _() how do I leave a line then??

**Comment by mridulnagpal** [comment]
@jvalleroy working now using linebreaks filter

**Comment by mridulnagpal** [comment]
@jvalleroy Is there anything else you wish me to add to it?

File: plinth/views.py [hunk 2]
```diff
@@ -38,19 +38,21 @@ def index(request):
     shortcuts = frontpage.get_shortcuts()
     selection = request.GET.get('selected')
 
-    details, details_label, configure_url = None, None, None
+    details, details_label, configure_url, icon_url = None, None, None, None
     if selection in frontpage.shortcuts:
         details = frontpage.shortcuts[selection]['details']
         details_label = frontpage.shortcuts[selection]['label']
         configure_url = frontpage.shortcuts[selection]['configure_url']
+        icon_url = frontpage.shortcuts[selection]['icon_url']
```
Comment (jvalleroy) [inline plinth/views.py]: I think this is not used anywhere, is it needed?

File: plinth/frontpage.py [hunk 3]
```diff
@@ -28,12 +28,15 @@ def get_shortcuts():
 
 
 def add_shortcut(id, label, url, icon, details=None, configure_url=None,
```
Comment (jvalleroy) [inline plinth/frontpage.py]: I think the old 'icon' parameter is no longer used, can it be removed?

File: plinth/frontpage.py [hunk 4]
```diff
@@ -28,12 +28,15 @@ def get_shortcuts():
...
-                 login_required=False):
+                 login_required=False, icon_url=None):
```
Comment (jvalleroy) [inline plinth/frontpage.py]: Since icon_url is no longer a URL, I suggest rename it to something like icon_name (or just icon if we remove the old one).

**Comment by jvalleroy** [comment]
@mridulnagpal Thanks, I've merged this, with some minor changes.

**Pull request closed without merging**